scalafix-testkit/src/main/scala/scalafix/testkit/SemanticRuleSuite.scala

```diff
@@ -98,13 +99,19 @@ abstract class SemanticRuleSuite(
         .collectFirst { case f if f.isFile => f.readAllBytes }
         .map(new String(_))
         .getOrElse {
-          if (Patch.isOnlyLintMessages(patch)) obtained // linter
-          else {
+          if (!Patch.isOnlyLintMessages(patch)) {
             val tried = candidateOutputFiles.mkString("\n")
             sys.error(
               s"""Missing expected output file for test ${diffTest.filename}. Tried:
                  |$tried""".stripMargin)
           }
+          if (!Patch.isAllAtomic(patch))
+            sys.error(
+              s"One or more rules used by test ${diffTest.filename} emit non-atomic " +
+                "top level patches. To fix this, please use `.atomic` on individual or group " +
+                "of patches that need to be atomically applied.")
+
+          obtained // linter
         }
```

Comment:
it needs to be "after line 108" (outside of the `.getOrElse` block and before the `val` below) 😉

Reply:
Indeed that doesn't make sense in the `.getOrElse` :)